### PR TITLE
Portfolio: preserve data in the browser

### PR DESCRIPTION
**Pull request by sheylaPozo**
- Use local storage to save user input.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -318,3 +319,37 @@ form.addEventListener("submit", (e) => {
...
+    localStorage.setItem('userInfo', userData);
+  }
+}
+
+form.addEventListener('input', (evt) => {
+  if (evt.target.dataset.id === 'name' || evt.target.dataset.id === 'email' || evt.target.dataset.id === 'message') {
+    saveData(evt.target.dataset.id, evt.target.value);
+  }
+});
+
```
Comment (Davidosky007) [inline script.js]: * When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data, currently your form doesn't do this please fix this as its part of the project requirement

File: script.js [hunk 2]
```diff
@@ -318,3 +319,37 @@ form.addEventListener("submit", (e) => {
...
+    const oldData = JSON.parse(localStorage.getItem('userInfo'));
+    const newData = {
+      ...oldData,
+      [key]: value
+    };
+    localStorage.setItem('userInfo', JSON.stringify(newData));
+  } else {
+    localStorage.setItem('userInfo', userData);
+  }
+}
```
Comment (Davidosky007) [inline script.js]: * Currently changes made on the input fields are not stored in local storage

File: script.js [hunk 3]
```diff
@@ -318,3 +319,37 @@ form.addEventListener("submit", (e) => {
...
+window.addEventListener('load', () => {
+  if (localStorage.getItem('userInfo')) {
+    const userInfo = JSON.parse(localStorage.getItem('userInfo'));
+    usernameInput.value = userInfo.name;
+    emailInput.value = userInfo.email;
+    messageText.value = userInfo.message;
+  } else {
+    localStorage.setItem('userInfo', userData);
+  }
+});
```
Comment (Davidosky007) [inline script.js]: * This error appears in the console when the input fields are populated 
* Please fix 🙏 
Screenshot (120)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @sheylaPozo, 

Good job so far!
* Nice job adding an extra reset button to clear the input field 💪 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
* I suggest your Pull Request have a descriptive title telling you what changes are inside and also a short summary describing introduced changes in more detail.
# _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: script.js (hunk 3 above)
Comment (sheylaPozo) [reply script.js]: _fixed the errors_

File: script.js (hunk 2 above)
Comment (sheylaPozo) [reply script.js]: ok corrected it

File: script.js (hunk 1 above)
Comment (sheylaPozo) [reply script.js]: ok corrected it

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @sheylaPozo,

Good job on implementing the suggested changes. Well Done.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**You may close this Pull Request and merge the changes**

  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
_**Dipesh Kumar**_

--------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**